### PR TITLE
feat [Infra-3051] use Eclipse Temurin for JDK distributions

### PR DESCRIPTION
**Pull request by dduportal**
This PR ensures that all the Docker images used by the Jenkins infrastructures are using Eclipse Temurin distribution for the JDK.

It introduces the following changes:

- Remove images that are not used or deprecated: coresdk(s), coreruntime, python(s), golang, terraform, docker, Maven Windows nanoserver
- [INFRA-3051] Use eclipse temurin (from upstream images, with a copy of the jlinked-JDK or default)
- Set JDK11 as default when no JDK version is explictly defined
- Define a static JDK installation path (`/opt/jdk-8`, `/opt-jdk11`) as per https://github.com/jenkins-infra/pipeline-library/issues/228#issuecomment-930219117 when possible

### CODE REVIEW

File: node/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,12 @@ FROM node:alpine
 
 ## Ignore apk package pinning as we always want the latest package here
 # hadolint ignore=DL3018
-RUN apk --no-cache add bash git openjdk8-jre
+RUN apk --no-cache add bash git
+
+# Install JDK from the official agent image
+ENV JAVA_HOME=/opt/jdk11
+ENV PATH "${JAVA_HOME}/bin:${PATH}"
+COPY --from=jnlp /opt/java/openjdk "${JAVA_HOME}"
```
Comment (timja) [inline node/Dockerfile:12]: how did this work before ? I don't see any java in this image =/

**Review by timja — APPROVED** [review]

File: maven/jdk11/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 FROM jenkins/inbound-agent:alpine as jnlp
 
-FROM maven:3.8-jdk-11-slim
+FROM maven:3.8.2-eclipse-temurin-11
```
Comment (timja) [inline maven/jdk11/Dockerfile:3]: does this image define a jdk in the location you want?

Comment (dduportal) [reply maven/jdk11/Dockerfile:3]: Not at all: these are the excpetion (the layers does not have the same order as the others still present).

Shall we install maven by ourselves instead? I mean, the value of using this official image is low: we only have to download a zip and extract it somewhere + update the PATH right?

**Comment by jglick** [comment]
> Define a static JDK installation path

It does not seem you are actually doing that in this PR, but anyway it would not be necessary: https://github.com/jenkins-infra/pipeline-library/issues/228#issuecomment-931493950

**Pull request merged**